compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -688,16 +689,17 @@ fn link_natively<'a, B: ArchiveBuilder<'a>>(
         out.extend(&output.stdout);
         let out = String::from_utf8_lossy(&out);
 
-        // Check to see if the link failed with "unrecognized command line option:
-        // '-no-pie'" for gcc or "unknown argument: '-no-pie'" for clang. If so,
-        // reperform the link step without the -no-pie option. This is safe because
-        // if the linker doesn't support -no-pie then it should not default to
-        // linking executables as pie. Different versions of gcc seem to use
-        // different quotes in the error message so don't check for them.
+        // Check to see if the link failed with an error message that indicates an
+        // unknown option. If so, reperform the link step without the -no-pie
+        // option. This is safe because if the linker doesn't support -no-pie then
+        // it should not default to linking executables as pie. Different versions
+        // of gcc seem to use different quotes in the error message so don't check
+        // for them.
+        let unknown_arg_regex =
+            Regex::new(r"(unknown|unrecognized) (command line )?(option|argument)").unwrap();
```

Comment:
This will recompile regex more than once

Reply:
Didn’t notice this was in a loop. Fixed